test/test.js

```diff
@@ -316,6 +316,34 @@ test('should send uncompressed if unsupported encoding', t => {
   })
 })
 
+test('should call callback if unsupported encoding', t => {
+  t.plan(3)
+  const fastify = Fastify()
+  fastify.register(compressPlugin, {
+    global: false,
+    onUnsupportedEncoding: (encoding, request, reply) => {
+      reply.code(406)
+      return JSON.stringify({ hello: encoding })
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.type('text/plain').compress(createReadStream('./package.json'))
+  })
+
+  fastify.inject({
+    url: '/',
+    method: 'GET',
+    headers: {
+      'accept-encoding': 'hello'
+    }
+  }, (err, res) => {
+    t.error(err)
+    t.strictEqual(res.statusCode, 406)
+    t.deepEqual(JSON.parse(res.payload), { hello: 'hello' })
+  })
+})
```

Comment:
Can you add a similar example where an error is thrown?

Reply:
Yes. Sorry for the delay. I was moving.